frontstage/controllers/survey_controller.py

```diff
@@ -30,26 +27,35 @@ def get_survey(survey_id):
     return response.json()
 
 
-def get_surveys_list(cases, party_id, list_type):
-    logger.info('Attempting to retrieve surveys list', party_id=party_id, list_type=list_type)
-
-    history_statuses = ['COMPLETE', 'COMPLETEDBYPHONE', 'NOLONGERREQUIRED']
-    if list_type == 'todo':
-        filtered_cases = [case
-                          for case in cases
-                          if case.get('caseGroup', {}).get('caseGroupStatus') not in history_statuses]
-    elif list_type == 'history':
-        filtered_cases = [case
-                          for case in cases
-                          if case.get('caseGroup', {}).get('caseGroupStatus') in history_statuses]
-    else:
-        logger.error('Invalid list type', party_id=party_id, list_type=list_type)
-        raise InvalidSurveyList(list_type)
-
-    surveys_data = [case_controller.build_full_case_data(case) for case in filtered_cases]
-    now = datetime.now(timezone.utc)
-    live_cases = [survey for survey in surveys_data if parse_date(survey['go_live']['timestamp']) < now]
-    enrolled_cases = [case for case in live_cases if case_controller.case_is_enrolled(case, party_id)]
-
-    logger.info('Successfully retrieved surveys list', party_id=party_id, list_type=list_type)
-    return enrolled_cases
+def get_survey_by_short_name(survey_short_name):
+    logger.debug('Attempting to retrieve survey by its short name', survey_short_name=survey_short_name)
+    url = f"{app.config['SURVEY_URL']}/surveys/shortname/{survey_short_name}"
+    response = requests.get(url, auth=app.config['SURVEY_AUTH'])
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        raise ApiError(logger, response,
+                       log_level='warning' if response.status_code == 404 else 'exception',
+                       message='Failed to retrieve survey by its short name',
+                       survey_short_name=survey_short_name)
+
+    logger.debug('Successfully retrieved survey by its short name', survey_short_name=survey_short_name)
+    return response.json()
+
+
+def get_enrolment_with_survey(enrolment, surveys):
+    logger.debug("Attempting to retrieve survey for enrolment", business_party=enrolment['business_party']['id'])
```

Comment:
Missing coverage for these new functions:

- get_enrolment_with_survey
- get_surveys_with_enrolments